### PR TITLE
feat: background search building

### LINKED ISSUE #1178: Build search indexes with background workers
When a schema is modified with existing documents and a search index is added we need to build the index in the background

### PR DESCRIPTION
**Pull request by garrensmith**
## Describe your changes
Enable background building of search indexes

## How best to test these changes
All tests should pass

## Issue ticket number and link
closes #1178

### CODE REVIEW

File: server/metadata/collection.go [hunk 1]
```diff
@@ -144,6 +144,22 @@ func (c *CollectionSubspace) Update(ctx context.Context, tx transaction.Tx, ns N
 
 	metadata.SearchState = newSearchState
 
+	if newSearchState == schema.SearchIndexWriteMode && config.DefaultConfig.Workers.Enabled {
```
Comment (himank) [inline server/metadata/collection.go:147]: Should we have `config.DefaultConfig.Workers.SearchEnabled`?

File: server/services/v1/workers/worker.go [hunk 2]
```diff
@@ -287,6 +296,68 @@ func (w *Worker) buildIndexTask(queueItem *metadata.QueueItem) error {
...
+	}
+	coll := db.GetCollection(task.CollName)
+
+	req := &api.BuildCollectionSearchIndexRequest{
+		Branch:     task.Branch,
+		Collection: task.CollName,
+		Project:    task.ProjName,
+	}
+
+	qr := database.NewQueryRunnerFactory(w.txMgr, nil, w.searchStore)
```
Comment (himank) [inline server/services/v1/workers/worker.go:329]: We can have this inside worker.

Comment (garrensmith) [reply server/services/v1/workers/worker.go:329]: I don't quite follow what you mean?

Comment (himank) [reply server/services/v1/workers/worker.go:333]: I mean we can create it when we are creating the worker.

**Review by himank — APPROVED** [review]

**Pull request merged**